lib/parallel_tests/grouper.rb

```diff
@@ -28,7 +28,39 @@ def in_even_groups_by_size(items, num_groups, options = {})
           raise 'Number of isolated processes must be less than total the number of processes'
         end
 
-        if isolate_count >= 1
+        if options[:specify_groups]
+          specify_test_process_groups = options[:specify_groups].split('|')
+          all_specified_tests = specify_test_process_groups.map { |group| group.split(',') }.flatten
+          if specify_test_process_groups.count > num_groups
+            raise 'Number of processes separated by pipe must be less than or equal to the total number of processes'
+          end
+
+          specified_items_found, items = items.partition do |item, _size|
+            all_specified_tests.any? { |specified_spec| item == specified_spec }
+          end
+
+          specified_specs_not_found = all_specified_tests - specified_items_found.map(&:first)
+          if specified_specs_not_found.any?
+            raise "Could not find #{specified_specs_not_found} from --specify-groups in the main selected files & folders"
+          end
+
+          if specify_test_process_groups.count == num_groups && items.flatten.any?
+            raise "The number of groups in --specify-groups matches the number of groups from -n but there were other specs found in the main selected files & folders not specified in --specify-groups. Make sure -n is larger than the number of processes in --specify-groups if there are other specs that need to be run. The specs that aren't run: #{items.map(&:first)}"
+          end
+
+          specify_test_process_groups.each_with_index do |specify_test_process, i|
+            groups[i] = specify_test_process.split(',')
+          end
+
+          return groups if specify_test_process_groups.count == num_groups
```

Comment:
should fail if there are no groups left, but tests left ?

Reply:
actually I think this return early is overkill with the early return above